### ISSUE TITLE
Origin and headers aren't passings to client

### ISSUE THREAD

**Issue by tsapko3628** [elpheria/rpc-websockets#105]
I'd want to use this client for test in node js environment. I tried to pass origin option (what exists in type declarations), but nothing happened. I also tried to pass headers and origin in headers, but nothing happened to. I found that it because only some options used and other options are ignored. It happening here https://github.com/elpheria/rpc-websockets/blob/master/src/lib/client.ts#L70. I don't want to fork it now so please fix it. Thank you.

**Comment by mkozjak** [comment]
Hello, you can fork it and provide a PR with your solution.

**Comment by beautyfree** [comment]
@tsapko3628 @mkozjak this feature is really needed

**Comment by trasherdk** [comment]
As I understand, there's no HTTP protocol, like headers, involved in a websocket connection, so exchange of stuff you normally would have in HTTP headers, like tokens and stuff, has to be handled by client/server conversation after a connection is made.
Take a look at [a-simple-example](https://javascript.info/websocket#a-simple-example) explaining the handshake thing.

**Comment by tsapko3628** [comment]
No. There is a http protocol and it uses for handshake and I can send http headers in handshake response. But this issue about another problem. Browser send http headers like "Origin" and "Cookie" in http handshake request. I want to use them in this package then I connect to server. I know that I can pass headers in browser (there're no APIs for it) but I still can do it on server side. My case is send origin and cookies then I writing tests. So, tests are running on server side and I can add some headers. Ws package allow to do it, types for rpc-websockets are also allow to do it, but some fields, which specified on types, haven't used. I just want to pass what to ws package. It's not feature request, it's bag report for rpc-websockets or types for rpc-websocket.
